test/common/secret/sds_api_test.cc

```diff
@@ -115,6 +115,92 @@ TEST_F(SdsApiTest, DynamicTlsCertificateUpdateSuccess) {
   handle->remove();
 }
 
+class PartialMockSds : public SdsApi {
+public:
+  PartialMockSds(NiceMock<Server::MockInstance>& server, Api::Api& api,
+                 NiceMock<Init::MockManager>& init_manager,
+                 envoy::api::v2::core::ConfigSource& config_source)
+      : SdsApi(
+            server.localInfo(), server.dispatcher(), server.random(), server.stats(),
+            server.clusterManager(), init_manager, config_source, "abc.com", []() {}, api) {}
+
+  MOCK_METHOD2(onConfigUpdate,
+               void(const Protobuf::RepeatedPtrField<ProtobufWkt::Any>&, const std::string&));
+  void setSecret(const envoy::api::v2::auth::Secret&) override {}
+  void validateConfig(const envoy::api::v2::auth::Secret&) override {}
+};
+
+// Basic test of delta's passthrough call to the state-of-the-world variant, to
+// increase coverage.
+TEST_F(SdsApiTest, Delta) {
+  Protobuf::RepeatedPtrField<envoy::api::v2::Resource> resources;
+  envoy::api::v2::auth::Secret secret;
+  secret.set_name("secret_1");
+  auto* resource = resources.Add();
+  resource->mutable_resource()->PackFrom(secret);
+  resource->set_name("secret_1");
+  resource->set_version("version1");
+
+  NiceMock<Server::MockInstance> server;
+  NiceMock<Init::MockManager> init_manager;
+  envoy::api::v2::core::ConfigSource config_source;
+  PartialMockSds sds(server, *api_, init_manager, config_source);
+  EXPECT_CALL(sds, onConfigUpdate(_, "version1"));
```

Comment:
Would there be value in using https://github.com/envoyproxy/envoy/blob/master/test/test_common/utility.h#L549 to expect the transform to take a particular form?

Reply:
Yeah, I originally had RepeatedProtoEq in there matching `resources`, forgetting that it was looking for an Any, not a Resource. I thought adding the wrapping that would make it work would be a bit much... although actually it turns out to have been pretty lightweight; I think I was just too tired when I was looking at it. Done.